app/features/counter/Counter.tsx

```diff
@@ -1,25 +1,14 @@
 import React from 'react';
+import { useSelector, useDispatch } from 'react-redux';
 import { Link } from 'react-router-dom';
 import styles from './Counter.css';
-import routes from '../constants/routes.json';
-
-type Props = {
-  increment: () => void;
-  incrementIfOdd: () => void;
-  incrementAsync: () => void;
-  decrement: () => void;
-  counter: number;
-};
-
-export default function Counter(props: Props) {
-  const {
-    increment,
-    incrementIfOdd,
-    incrementAsync,
-    decrement,
-    counter
-  } = props;
+import routes from '../../constants/routes.json';
+import { RootState } from '../../store';
+import { counterActions } from './counterSlice';
```

Comment:
Honestly, I see added value exporting `counterActions` because `counterSlice` could export a lot of things which is _not_ an actions.  But even it is exporting only actions, when I starting to type `dispatch(counterActions.` ctrl-space, I immediately got a list of available actions for counter.  Without this I have a plenty of functions available which can be actions or not(e.g. I can imagine `selectUser` as an action, but it is perfectly fine as a selector too.

Reply:
What I'm suggesting is just `import {actionA, actionB} from "./someSlice"`.

Reply:
It was clear what you suggest, I just tried to defend `counterActions` which I found a bit helpful on the first try. But I admit there is also some issues with that approach, so its gone:)
But now could you please take a look on tests, spying actions feels a bit weird now, it was much cleaner when I got that object where I could freely spy individual action creators:/